suppression-backup-service/exporter/exporter.go

```diff
@@ -0,0 +1,204 @@
+	}
+	defer func() {
+		_ = tempFile.Close()
+		_ = os.Remove(tempFile.Name())
+	}()
+
+	// export to temp file
+	err = repo.Backup(tempFile)
+	if err != nil {
+		return fmt.Errorf("error exporting suppression list: %w", err)
+	}
+
+	// move temp file to final destination
+	file.Mu.Lock()
+	defer file.Mu.Unlock()
+	err = os.Rename(tempFile.Name(), file.Path)
+	if err != nil {
+		return fmt.Errorf("error moving temp file to final destination: %w", err)
+	}
+
+	return nil
+}
+
+func newBadgerDBInstance(directoryName string, pkgLogger logger.Logger) (suppression.Repository, string, error) {
+	tmpDir, err := misc.CreateTMPDIR()
+	if err != nil {
+		return nil, "", fmt.Errorf("could not create tmp dir: %w", err)
+	}
+
+	var baseDir string
+	if directoryName == "" {
+		directoryName = path.Join(tmpDir, strings.ReplaceAll(uuid.New().String(), "-", ""))
+	}
+
+	baseDir, err = os.MkdirTemp(tmpDir, directoryName)
+	if err != nil {
+		return nil, "", fmt.Errorf("could not create tmp dir: %w", err)
+	}
+
+	repository, err := suppression.NewBadgerRepository(
+		baseDir,
+		pkgLogger)
+	if err != nil {
+		return nil, "", fmt.Errorf("could not create badger repository: %w", err)
+	}
+	return repository, baseDir, nil
+}
+
+type Exporter struct {
+	id             identity.Identifier
+	file           model.File
+	log            logger.Logger
+	pollIntervalFn func() time.Duration
+}
+
+// WithPollIntervalFn sets the interval at which the syncer will poll the backend
+func WithPollIntervalFn(pollIntervalFn func() time.Duration) ExporterOpt {
+	return func(c *Exporter) {
+		c.pollIntervalFn = pollIntervalFn
+	}
+}
+
+// NewExporter creates a new syncer
+func NewExporter(id identity.Identifier, file model.File, log logger.Logger, opts ...ExporterOpt) (*Exporter, error) {
+	e := &Exporter{
+		id:             id,
+		file:           file,
+		log:            log,
+		pollIntervalFn: func() time.Duration { return 30 * time.Second },
+	}
+
+	for _, opt := range opts {
+		opt(e)
+	}
+	return e, nil
+}
+
+func (e *Exporter) FullExporterLoop(ctx context.Context) error {
+	var interval time.Duration
+	config.RegisterDurationConfigVariable(3600, &interval, true, time.Second, "SuppressionExporter.fullExportInterval")
+
+	repo, _, err := newBadgerDBInstance("fullsuppression", e.log)
+	if err != nil {
+		return err
+	}
+	syncer, err := suppression.NewSyncer(
+		config.GetString("SUPPRESS_USER_BACKEND_URL", "https://api.rudderstack.com"),
+		e.id,
+		repo,
+		suppression.WithLogger(e.log),
+		suppression.WithHttpClient(&http.Client{Timeout: config.GetDuration("HttpClient.suppressUser.timeout", 30, time.Second)}),
+		suppression.WithPageSize(config.GetInt("BackendConfig.Regulations.pageSize", 5000)),
+	)
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(e.pollIntervalFn()):
+			err := syncer.Sync(ctx)
+			if err != nil {
+				e.log.Errorf("error syncing suppression list: %v", err)
```

Comment:
the reason behind logging the error and not returning was, I thought that because of some issue the sync failed. So, maybe the next time it won't. But, I think what you are saying make sense. Since, assuming that in next retry after a while it will succeed isn't appropriate. And, we should assuming that the Sync func returned error only after some retry. And, hence we should not ignore it. As ignoring might raise a scenario where sync always throw error and hence nothing is ever stored and exported.